app/models/candidate_interface/personal_details_form.rb

```diff
@@ -70,15 +70,17 @@ def date_of_birth
 
       if valid_year?(year) && Date.valid_date?(*date_args)
         Date.new(*date_args)
+      else
+        OpenStruct.new(day: day, month: month, year: year)
```

Comment:
How about a plain `Struct`? The keys never change

Reply:
It causes `unknown keywords: day, month, year` error

Reply:
`Struct` and `OpenStruct` have different interfaces. The equivalent `Struct` code would be 

```
Struct.new(:day, :month, :year).new(day, month, year)
```

Reply:
Thanks, I changed this in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1003/commits/54c327762f421c982e7db498761bf5a4faae9107